packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -288,48 +277,192 @@ class CupertinoDialogAction extends StatelessWidget {
+  final Widget title;
 
-class _CupertinoButtonBar extends StatelessWidget {
-  const _CupertinoButtonBar({
+  /// The (optional) content of the dialog is displayed in the center of the
+  /// dialog in a lighter font.
+  ///
+  /// Typically a [Text] widget.
+  final Widget message;
+
+  /// A scroll controller that can be used to control the scrolling of the
+  /// message in the dialog.
+  ///
+  /// Defaults to null, and is typically not needed, since most alert messages
+  /// are short.
+  final ScrollController scrollController;
+
+  @override
+  Widget build(BuildContext context) {
+    final List<Widget> titleContentGroup = <Widget>[];
+    if (title != null) {
+      titleContentGroup.add(new Padding(
+        padding: new EdgeInsets.only(
+          left: _kEdgePadding,
+          right: _kEdgePadding,
+          bottom: message == null ? _kEdgePadding : 1.0,
+          top: _kEdgePadding,
+        ),
+        child: new DefaultTextStyle(
+          style: _kCupertinoDialogTitleStyle,
+          textAlign: TextAlign.center,
+          child: title,
+        ),
+      ));
+    }
+
+    if (message != null) {
+      titleContentGroup.add(
+        new Padding(
+          padding: new EdgeInsets.only(
+            left: _kEdgePadding,
+            right: _kEdgePadding,
+            bottom: _kEdgePadding,
+            top: title == null ? _kEdgePadding : 1.0,
+          ),
+          child: new DefaultTextStyle(
+            style: _kCupertinoDialogContentStyle,
+            textAlign: TextAlign.center,
+            child: message,
+          ),
+        ),
+      );
+    }
+
+    if (titleContentGroup.isEmpty) {
+      return new SingleChildScrollView(
+        controller: scrollController,
+        child: new Container(width: 0.0, height: 0.0),
+      );
+    }
+
+    // Add padding between the widgets if necessary.
+    if (titleContentGroup.length > 1) {
+      titleContentGroup.insert(1, const Padding(padding: const EdgeInsets.only(top: 8.0)));
+    }
+
+    return new Flexible(
+      flex: 3,
+      child: new CupertinoScrollbar(
+        child: new SingleChildScrollView(
+          controller: scrollController,
+          child: new Column(
+            mainAxisSize: MainAxisSize.max,
+            crossAxisAlignment: CrossAxisAlignment.stretch,
+            children: titleContentGroup,
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+
+/// An Action Items section typically used in a [CupertinoAlertDialog].
+///
+/// If _shouldLayoutActionsVertically(_) is true, they are laid out vertically
+/// in a column; else they are laid out horizontally in a row. If there isn't
+/// enough room to show all the [children] vertically, they are wrapped in a
+/// [CupertinoScrollbar] widget. If [children] is null or empty, it returns null.
+class _CupertinoAlertActionSection extends StatelessWidget {
+  const _CupertinoAlertActionSection({
     Key key,
     this.children,
+    this.scrollController,
   }) : super(key: key);
```

Comment:
assert children != null, since you already guarantee that above

Reply:
added